### PR TITLE
Update Connector and Test to typescript

### PR DESCRIPTION
**Pull request by prodesert22**

### CODE REVIEW

**Review by benber86 — APPROVED** [review]

File: contracts/avalanche/connectors/pangolin/staking/helpers.sol [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+     * @dev Pangolin Token
+     */
+    IERC20 internal constant PNG = IERC20(0x60781C2586D68229fde47564546784ab3fACA982);
+
+    // LP Staking, use minichefv2 to staking lp tokens and earn png
+    function _depositLPStake(
+        uint pid,
+        uint amount
+    ) internal returns (address lpTokenAddr) {
+        require(pid <= minichefv2.poolLength(), "Invalid pid!");
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/helpers.sol]: poolLength() returns the number of pools which is a 0-based index. for instance right now there are 53 pools, but the highest pool ID is 52. this should require `pid < minichefv2.poolLength()`

File: contracts/avalanche/connectors/pangolin/staking/main.sol [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        _eventName = "LogExitPNGStake(address,uint256,uint256,address)";
+        _eventParam = abi.encode(stakingContract, exitAmount, rewardAmount, rewardToken);
+    }
+
+    function getRewardPNGstake(
+        address stakingContract
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        (uint256 rewardAmount, address rewardToken) = _getsingleStakeReward(stakingContract);
+
+        _eventName = "LogGetRewardtPNGStake(address,uint256,address)";
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/main.sol]: is this the correct event name?

File: contracts/avalanche/connectors/pangolin/staking/helpers.sol [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+        minichefv2.deposit(pid, amount, address(this));
+        lpTokenAddr = address(lptoken);
+    }
+
+    function _withdraw_LP_Stake(
+        uint pid,
+        uint amount
+    ) internal returns (address lpTokenAddr) {
+        require(pid <= minichefv2.poolLength(), "Invalid pid!");
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/helpers.sol]: same concern with the 0-based index

File: contracts/avalanche/connectors/pangolin/staking/helpers.sol [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+        IERC20 lptoken = minichefv2.lpToken(pid);
+        lpTokenAddr = address(lptoken);
+    }
+
+    function _withdraw_and_getRewards_LP_Stake(
+        uint pid,
+        uint amount
+    ) internal returns (uint256 rewardAmount, address lpTokenAddr) {
+        require(pid <= minichefv2.poolLength(), "Invalid pid!");
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/helpers.sol]: same concern with the 0-based index

File: contracts/avalanche/connectors/pangolin/staking/helpers.sol [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        minichefv2.withdrawAndHarvest(pid, amount, address(this));
+
+        IERC20 lptoken = minichefv2.lpToken(pid);
+        lpTokenAddr = address(lptoken);
+    }
+
+    function _getLPStakeReward(
+        uint pid
+    ) internal returns (uint256 rewardAmount, address lpTokenAddr) {
+        require(pid <= minichefv2.poolLength(), "Invalid pid!");
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/helpers.sol]: same concern with the 0-based index

File: contracts/avalanche/connectors/pangolin/staking/helpers.sol [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        minichefv2.harvest(pid, address(this));
+
+        IERC20 lptoken = minichefv2.lpToken(pid);
+        lpTokenAddr = address(lptoken);
+    }
+
+    function _emergencyWithdraw_LP_Stake(
+        uint pid
+    ) internal returns (uint256 lpAmount, address lpTokenAddr) {
+        require(pid <= minichefv2.poolLength(), "Invalid pid!");
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/helpers.sol]: same concern with the 0-based index

File: contracts/avalanche/connectors/pangolin/staking/helpers.sol [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+        exitAmount = stakingContract.balanceOf(address(this));
+        rewardAmount = stakingContract.rewards(address(this));
+
+        require(exitAmount > 0, "No balance to exit");
+
+        stakingContract.exit();
+    }
+
+    function _getsingleStakeReward(
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/helpers.sol]: nit picking, but should this function name be camel cased to match the other formatting?

File: contracts/avalanche/connectors/pangolin/staking/main.sol [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        _eventName = "LogWithdrawLpAndClaim(address,uint256,uint256,uint256,uint256,uint256)";
+        _eventParam = abi.encode(lpTokenAddr, pid, _amt, rewardAmount, getId, setId);
+    }
+
+    function claimLpRewards(
+        uint pid
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        (uint256 rewardAmount, address lpTokenAddr) = _getLPStakeReward(pid);
+
+        _eventName = "LogClaimLpReward(address,uint256,uint256,)";
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/main.sol]: is this trailing comma intentional?

File: contracts/avalanche/connectors/pangolin/staking/main.sol [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+    function exitPNGStake(
+        address stakingContract
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        (uint256 exitAmount, uint256 rewardAmount, address rewardToken) = _exitSingleStake(stakingContract);
+
+        _eventName = "LogExitPNGStake(address,uint256,uint256,address)";
+        _eventParam = abi.encode(stakingContract, exitAmount, rewardAmount, rewardToken);
+    }
+
+    function getRewardPNGstake(
```
Comment (bmino) [inline contracts/avalanche/connectors/pangolin/staking/main.sol]: camel cased function name here as getRewardPNGStake ?

File: contracts/avalanche/connectors/pangolin/staking/helpers.sol (hunk 1 above)
Comment (prodesert22) [reply contracts/avalanche/connectors/pangolin/staking/helpers.sol]: Ok

File: contracts/avalanche/connectors/pangolin/staking/main.sol (hunk 8 above)
Comment (prodesert22) [reply contracts/avalanche/connectors/pangolin/staking/main.sol]: No, I forgot to remove it 😅

File: contracts/avalanche/connectors/pangolin/staking/main.sol (hunk 9 above)
Comment (prodesert22) [reply contracts/avalanche/connectors/pangolin/staking/main.sol]: I rename this function to claimPNGStakeReward to look similar to the event and with the function of claimLpRewards

**Review by bmino — APPROVED** [review]

**Pull request merged**